src/main/kotlin/org/jetbrains/intellij/platform/gradle/tasks/JarSearchableOptionsTask.kt

```diff
@@ -80,8 +75,14 @@ abstract class JarSearchableOptionsTask : Jar() {
 
                 from(prepareJarSearchableOptionsTask)
 
+                dependsOn(prepareJarSearchableOptionsTask)
+
+                val prepareOptionsTaskEnabled = prepareJarSearchableOptionsTask.get().enabled
+                inputs.property("prepareOptionsTaskEnabled", prepareOptionsTaskEnabled)
+                inputs.property("intellijPlatform.buildSearchableOptions", buildSearchableOptionsEnabled)
+
                 onlyIf {
-                    buildSearchableOptionsEnabled.get()
+                    prepareOptionsTaskEnabled && buildSearchableOptionsEnabled.get()
```

Comment:
Please note that this is different from how it was. I added an explicit dependency between buildSearchableOptions -> prepareJarSearchableOptions -> jarSearchableOptions instead of all of them relying on  buildSearchableOptions.